pkg/kubelet/network/cni/cni.go

```diff
@@ -294,5 +326,24 @@ func buildCNIRuntimeConf(podName string, podNs string, podInfraContainerID kubec
 		},
 	}
 
+	// port mappings are a cni capability-based args, rather than parameters
+	// to a specific plugin
+	portMappings, err := plugin.host.GetPodPortMappings(podInfraContainerID.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve port mappings: %v", err)
+	}
+	portMappingsParam := make([]cniPortMapping, 0, len(portMappings))
+	for _, p := range portMappings {
+		portMappingsParam = append(portMappingsParam, cniPortMapping{
+			HostPort:      p.HostPort,
+			ContainerPort: p.ContainerPort,
+			Protocol:      strings.ToLower(string(p.Protocol)),
+			HostIP:        p.HostIP,
+		})
+	}
+	rt.CapabilityArgs = map[string]interface{}{
```

Comment:
@feiskyer we don't intend to put that capability into the main plugins.  Instead, you would create a CNI ConfigList that includes the CNI "portmap" plugin here https://github.com/containernetworking/plugins/pull/1.  When the Kube CNI driver calls AddNetworkList(), libcni will execute each plugin in the list sequentially, passing the results from each plugin along to the next.  The "portmap" plugin consumes that result and sets up the mapings as required.